java/src/main/java/ai/rapids/cudf/ArrowColumnBuilder.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ *  Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+/**
+ * Column builder from Arrow data. This builder takes in pointers to the Arrow off heap
+ * memory and allows efficient building of CUDF ColumnVectors from that arrow data.
+ * The caller can add multiple batches where each batch corresponds to Arrow data
+ * and those batches get concatenated together after being converted to CUDF
+ * ColumnVectors.
+ */
+public final class ArrowColumnBuilder implements AutoCloseable {
+    private DType type;
+    private ArrayList<Long> data = new ArrayList<>();
+    private ArrayList<Long> dataLength = new ArrayList<>();
+    private ArrayList<Long> validity = new ArrayList<>();
+    private ArrayList<Long> validityLength = new ArrayList<>();
+    private ArrayList<Long> offsets = new ArrayList<>();
+    private ArrayList<Long> offsetsLength = new ArrayList<>();
+    private ArrayList<Long> nullCount = new ArrayList<>();
+    private ArrayList<Long> rows = new ArrayList<>();
+    private int numBatches = 0;
+    private String colName;
+
+    public ArrowColumnBuilder(HostColumnVector.DataType type, String name) {
+      this.type = type.getType();
```

Comment:
If we don't support nested types then we should throw an exception here if we get one, and add it to the java docs.

Reply:
the Jni layer will end up throwing later when it tries to build it. I didn't want to create the same case in 2 places